### PR TITLE
Add email check API call to section instructor controller

### PR DESCRIPTION
**Pull request by etaderhold**
This adds a new API endpoint to check whether an email address corresponds to a teacher on our platform. This is necessary for the co-teacher interface that allows inviting users by email address.

The API includes an optional `section_id` parameter that can be included when calling the API from the section edit (rather than create) interface. When this parameter is included the API will additionally check whether the section is at capacity and whether the teacher has already been invited to teach that section.

## Links

- [JIRA](https://codedotorg.atlassian.net/browse/TEACH-706)

## Testing story

Automated tests included

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/controllers/api/v1/section_instructors_controller_test.rb [hunk 1]
```diff
@@ -165,6 +165,48 @@ class Api::V1::SectionInstructorsControllerTest < ActionController::TestCase
     put :decline, params: {id: si.id}
     assert_response :forbidden
   end
+
+  test 'logged-out user cannot check email addresses' do
```
Comment (lfryemason) [inline dashboard/test/controllers/api/v1/section_instructors_controller_test.rb:169]: Test case for > 5 co-teachers?

File: dashboard/test/controllers/api/v1/section_instructors_controller_test.rb [hunk 2]
```diff
@@ -165,6 +165,48 @@ class Api::V1::SectionInstructorsControllerTest < ActionController::TestCase
...
+    assert_response :success
+  end
+
+  test 'teacher gets error checking own email address' do
+    sign_in @teacher
+    get :check, params: {email: @teacher.email}
+    assert_response :bad_request
+  end
+
+  test 'teacher gets error checking non-user email address' do
```
Comment (lfryemason) [inline dashboard/test/controllers/api/v1/section_instructors_controller_test.rb:186]: Test case for checking gets error checking a student email address?

**Review by lfryemason — APPROVED** [review]

File: dashboard/test/controllers/api/v1/section_instructors_controller_test.rb (hunk 1 above)
Comment (etaderhold) [reply dashboard/test/controllers/api/v1/section_instructors_controller_test.rb:169]: Done.

File: dashboard/test/controllers/api/v1/section_instructors_controller_test.rb (hunk 2 above)
Comment (etaderhold) [reply dashboard/test/controllers/api/v1/section_instructors_controller_test.rb:186]: We don't actually store student email addresses. https://blog.code.org/post/147756946588/codeorgs-new-login-approach-to-student-privacy

**Pull request merged**